SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,31 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // it will retrun a node. if it has only undummy child node, it will return that child node, else it will return itself
+    func mergeNode() -> Node {
+        var unDummyNum = 0
+        var unDummyChildren = Node()
```

Comment:
So two things here. Since you are only storing one child in this variable I think you should change the 'children' to 'child'. The second thing is that, in English, you can't stick 'un' on 'dummy'. You need to use 'non' here. I have no idea why. English is weird. So I think the name of this variable should actually be `nonDummyChild`.